vm/actor/src/builtin/puppet/mod.rs

```diff
@@ -0,0 +1,217 @@
+    {
+        Err(ser::Error::custom(
+            "Automatic fail when serializing FailToMarshalCBOR",
+        ))
+    }
+}
+
+impl<'de> Deserialize<'de> for FailToMarshalCBOR {
+    fn deserialize<D>(_deserializer: D) -> Result<Self, <D as Deserializer<'de>>::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        Err(de::Error::custom(
+            "Automatic fail when deserializing FailToMarshalCBOR",
+        ))
+    }
+}
+
+impl Cbor for FailToMarshalCBOR {}
+
+#[derive(Default, Serialize_tuple, Deserialize_tuple)]
+pub struct State {
+    opt_fail: Vec<Option<FailToMarshalCBOR>>,
+}
+
+impl Cbor for State {}
+
+pub struct Actor;
+
+impl Actor {
+    fn constructor<BS, RT>(rt: &mut RT) -> Result<(), ActorError>
+    where
+        BS: BlockStore,
+        RT: Runtime<BS>,
+    {
+        rt.validate_immediate_caller_accept_any()?;
+
+        rt.create(&State::default())?;
+        Ok(())
+    }
+
+    fn send<BS, RT>(rt: &mut RT, params: SendParams) -> Result<SendReturn, ActorError>
+    where
+        BS: BlockStore,
+        RT: Runtime<BS>,
+    {
+        rt.validate_immediate_caller_accept_any()?;
+
+        let res = rt.send(
+            params.to,
+            params.method,
+            Serialized::serialize(params.params).unwrap(),
+            params.value,
+        );
+
+        let return_bytes = res.map_err(|_| {
+            ActorError::new(
+                ExitCode::ErrIllegalState,
+                "failed to get return bytes".to_string(),
+            )
+        })?;
+
+        Ok(SendReturn {
+            return_bytes,
+            code: ExitCode::Ok,
+        })
+    }
+
+    fn send_marshal_cbor_failure<BS, RT>(
+        rt: &mut RT,
+        params: SendParams,
+    ) -> Result<SendReturn, ActorError>
+    where
+        BS: BlockStore,
+        RT: Runtime<BS>,
+    {
+        rt.validate_immediate_caller_accept_any()?;
+
+        let res = rt.send(
+            params.to,
+            params.method,
+            Serialized::serialize(FailToMarshalCBOR::default()).unwrap(),
+            params.value,
+        );
+
+        let return_bytes = res.map_err(|_| {
+            ActorError::new(
+                ExitCode::ErrIllegalState,
+                "failed to get return bytes".to_string(),
+            )
+        })?;
+
+        Ok(SendReturn {
+            return_bytes,
+            code: ExitCode::Ok,
+        })
+    }
+
+    fn return_marshal_cbor_failure<BS, RT>(rt: &mut RT) -> Result<FailToMarshalCBOR, ActorError>
+    where
+        BS: BlockStore,
+        RT: Runtime<BS>,
+    {
+        rt.validate_immediate_caller_accept_any()?;
+        Ok(FailToMarshalCBOR::default())
+    }
+
+    fn runtime_transaction_marshal_cbor_failure<BS, RT>(rt: &mut RT) -> Result<(), ActorError>
+    where
+        BS: BlockStore,
+        RT: Runtime<BS>,
+    {
+        rt.validate_immediate_caller_accept_any()?;
+
+        rt.transaction(|st: &mut State, _| {
+            st.opt_fail = vec![];
```

Comment:
I don't think this fails in our case (and hope this doesn't ever get hit in our node outside of this, but I don't see why theirs would fail if that type never gets serialized) You might want to write a test or something to check if serializing an empty vec of `FailToMarshalCBOR` will error

Reply:
Just wrote the test and a empty vec of FailToMarshalCBOR can be successfully serialized